tests/py/test_boost_python.py

```diff
@@ -5,6 +5,30 @@
 	# If the import fails, import the knowrob.so directly
 	from knowrob import *
 
+def perform_query(settings_path, query_string,):
+	# Helper function to perform a query on a knowledge base
+	# Load the settings
+	modalities = {
+		"epistemicOperator": EpistemicOperator.KNOWLEDGE,
+		"aboutAgentIRI": "",
+		"confidence": 0.0,
+		"temporalOperator": TemporalOperator.CURRENTLY,
+		"minPastTimestamp": -1.0,
+		"maxPastTimestamp": -1.0,
+	}
+	# Load the settings
+	kb = KnowledgeBase(settings_path)
+	# Create a formula for the query
+	phi = QueryParser.parse(query_string)
+	# Apply the modality
+	mPhi = applyModality(modalities, phi)
```

Comment:
Why don't you put the modal operator into the query string instead? Then you do not need to apply a modal operator after parsing.

Also there is already a syntax available for doing this. You can find an example here: https://github.com/knowrob/knowrob/blob/dev/tests/KnowledgeBaseTest.cpp#L273